### PR TITLE
add Delete/Overwrite named regions functionality v2

### PR DESCRIPTION
**Pull request by bthieurmel**
Without the white space changes...!

### CODE REVIEW

File: R/wrappers.R [hunk 1]
```diff
@@ -2809,7 +2814,25 @@ createNamedRegion <- function(wb, sheet, cols, rows, name) {
...
+    stop("First argument must be a Workbook.")
+  }
+  
+  ex_names <- regmatches(wb$workbook$definedNames, regexpr('(?<=name=")[^"]+', wb$workbook$definedNames, perl = TRUE))
+  ex_names <- tolower(replaceXMLEntities(ex_names))
+  
+  if (tolower(name) %in% ex_names) {
+    wb$workbook$definedNames <- wb$workbook$definedNames[!ex_names %in% tolower(name)]
+  } else {
+    warning(sprintf("Canno't  find Named region with name '%s'", name))
```
Comment (JanMarvin) [inline R/wrappers.R]: Typo

File: R/wrappers.R [hunk 2]
```diff
@@ -2784,13 +2786,16 @@ createNamedRegion <- function(wb, sheet, cols, rows, name) {
   ex_names <- regmatches(wb$workbook$definedNames, regexpr('(?<=name=")[^"]+', wb$workbook$definedNames, perl = TRUE))
   ex_names <- tolower(replaceXMLEntities(ex_names))
 
-  if (tolower(name) %in% ex_names) {
-    stop(sprintf("Named region with name '%s' already exists!", name))
-  } else if (grepl("^[A-Z]{1,3}[0-9]+$", name)) {
+  if (tolower(name) %in% ex_names & !overwrite) {
+    stop(sprintf("Named region with name '%s' already exists! Use overwrite  = TRUE if you want to replace it", name))
```
Comment (JanMarvin) [inline R/wrappers.R:2796]: double whitespace

File: R/wrappers.R [hunk 3]
```diff
@@ -2809,7 +2814,25 @@ createNamedRegion <- function(wb, sheet, cols, rows, name) {
 }
 
 
-
+#' @export
```
Comment (JanMarvin) [inline R/wrappers.R:2823]: Could you add some text to the function for the man-page. What it does and and an example how the code runs. Currently it fails the check.

**Review by JanMarvin — CHANGES_REQUESTED** [review]
Looks good, minor changes, but please add the missing documentation and a test.

**Comment by bthieurmel** [comment]
Done

File: R/wrappers.R [hunk 4]
```diff
@@ -2809,7 +2820,25 @@ createNamedRegion <- function(wb, sheet, cols, rows, name) {
...
+    stop("First argument must be a Workbook.")
+  }
+  
+  ex_names <- regmatches(wb$workbook$definedNames, regexpr('(?<=name=")[^"]+', wb$workbook$definedNames, perl = TRUE))
+  ex_names <- tolower(replaceXMLEntities(ex_names))
+  
+  if (tolower(name) %in% ex_names) {
+    wb$workbook$definedNames <- wb$workbook$definedNames[!ex_names %in% tolower(name)]
+  } else {
+    warning(sprintf("Canno't find Named region with name '%s'", name))
```
Comment (JanMarvin) [inline R/wrappers.R]: should be "Cannot",  "Can not" or "Can't" :smile:

**Comment by JanMarvin** [comment]
Thanks!

**Review by JanMarvin — APPROVED** [review]

**Pull request merged**